RxCocoa/Foundation/NSObject+Rx.swift

```diff
@@ -95,6 +145,50 @@ extension Reactive where Base: NSObject {
                 return n as? E
             }
     }
+
+    /**
+     Observes values on `keyPath` starting from `self` with `options` and doesn't retain `self`.
+
+     It can be used in all cases where `observe` can be used and additionally
+
+     * because it won't retain observed target, it can be used to observe arbitrary object graph whose ownership relation is unknown
+     * it can be used to observe `weak` properties
+
+     **Since it needs to intercept object deallocation process it needs to perform swizzling of `dealloc` method on observed object.**
+
+     - parameter keyPath: Key path of property names to observe.
+     - parameter options: KVO mechanism notification options.
+     - returns: Observable sequence of objects on `keyPath`.
+     */
+    @available(swift 4.0)
+    public func observeWeakly<E>(_ keyPath: KeyPath<Base, E>, options: KeyValueObservingOptions = [.new, .initial]) -> Observable<E?> {
+        guard let keyPathString = keyPath._kvcKeyPathString else {
+            rxFatalError("Key path cannot be observed. You may need to prefix it with @objc.")
+        }
+        return observeWeakly(E.self, keyPathString, options: options)
+    }
+
+    /**
+     Observes values on `keyPath` starting from `self` with `options` and doesn't retain `self`.
+
+     It can be used in all cases where `observe` can be used and additionally
+
+     * because it won't retain observed target, it can be used to observe arbitrary object graph whose ownership relation is unknown
+     * it can be used to observe `weak` properties
+
+     **Since it needs to intercept object deallocation process it needs to perform swizzling of `dealloc` method on observed object.**
+
+     - parameter keyPath: Key path of property names to observe.
+     - parameter options: KVO mechanism notification options.
+     - returns: Observable sequence of objects on `keyPath`.
+     */
+    @available(swift 4.0)
+    public func observeWeakly<E>(_ keyPath: KeyPath<Base, E?>, options: KeyValueObservingOptions = [.new, .initial]) -> Observable<E?> {
+        guard let keyPathString = keyPath._kvcKeyPathString else {
+            rxFatalError("Key path cannot be observed. You may need to prefix it with @objc.")
+        }
```

Comment:
Instead of repeating 
```
guard let keyPathString = keyPath._kvcKeyPathString else {
     rxFatalError("Key path cannot be observed. You may need to prefix it with @objc.")
}
```

It might be better to create extension on `KeyPath` so we can do something like:
```
return observeWeakly(E.self, keyPath.kvcKeyPathString, options: options)
```

```
extension KeyPath {
     var kvcKeyPathString: String {
           guard let keyPathString = keyPath._kvcKeyPathString else {
               rxFatalError("Key path cannot be observed. You may need to prefix it with @objc.")
           }
           return keyPathString
     } 
}
```

and thus avoid duplication of logic.